src/efftester.ml

```diff
@@ -221,22 +252,26 @@ let eff_leq eff eff_exp =
 
 let rec occurs tvar = function
   | Typevar a -> tvar = a
+  | Option a -> occurs tvar a
   | List a -> occurs tvar a
   | Fun (a, _, b) -> occurs tvar a || occurs tvar b
-  | _ -> false
+  | Unit | Int | Float | Bool | String -> false
```

Comment:
Thanks for making these matchings exhaustive (here and below), it's a very good idea!